src/backend/drm/compositor/mod.rs

```diff
@@ -1035,6 +1035,59 @@ bitflags::bitflags! {
     }
 }
 
+bitflags::bitflags! {
+    /// Possible capabilities
+    #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+    pub struct Capabilities: u32 {
+        /// Explicit sync
+        const EXPLICIT_SYNC = 1;
+    }
+}
+
+impl Capabilities {
+    /// Get the supported capabilities of the provided surface
+    pub fn from_surface(surface: &DrmSurface) -> Result<Self, DrmError> {
+        let mut capabilities = Capabilities::empty();
+
+        let driver = surface.get_driver().map_err(|err| {
+            DrmError::Access(AccessError {
+                errmsg: "Failed to query drm driver",
+                dev: surface.dev_path(),
+                source: err,
+            })
+        })?;
+
+        // `IN_FENCE_FD` makes commit fail on Nvidia driver
+        // https://github.com/NVIDIA/open-gpu-kernel-modules/issues/622
+        let is_nvidia = driver.name().to_string_lossy().to_lowercase().contains("nvidia")
+            || driver
+                .description()
+                .to_string_lossy()
+                .to_lowercase()
+                .contains("nvidia");
```

Comment:
why not make this a debug flag or environment variable instead, like `SMITHAY_DEBUG_DISABLE_EXPLICIT_SYNC`? I don't think it should be on the compositor to deal with misbehaving drivers in this case.

Reply:
I find env vars have their own ux issues since many people don't know of the probably dozen that exist which you could tune.

Also an explicit type (with a little more code) means you could switch at runtime between modes.

I could see the env var make sense for configuring the default value.

Reply:
Both sounds good to me, if it can be shimmied in here.

Reply:
I am not a big fan of environment variables when they are avoidable. Giving explicit control allows projects like niri to implement thst config option in a way more visible way.

The reason for keeping the nvidia check is simply to not break existing code.

But I am not opposed to adding an environment variable that overrides the default.